### PR TITLE
#164069218 send verification email after successful signup

### PR DESCRIPTION
**Pull request by CryceTruly**
### What does this PR do?

- add EmailVerifyAPIView
- implement token decoding
- add URL for EmailVerifyAPIView
- write tests for the EmailVerifyAPIView

### Description of Task to be completed?
- Users should receive an email containing an activation link when they successfully signup.
- Users' accounts should be activated when they click on the activation link they received in their email

#### How should this be manually tested?
```
- clone the repo
- cd Ah-backend-aquaman
- make a virtual environment and activate it
- install packages with pip install -r requirements.txt
- run python manage.py runserver
- create an account using a rest client by hitting POST /api/users/
- Try signing in by hitting POST /api/users/login/
- After it fails, go to your mailbox and click on the link to verify
- Then Try signing in again
```

### Any background context you want to provide?
With email verification, we prevent malicious bots from creating random accounts and creating DOS attacks on our server.

### What are the relevant pivotal tracker stories?
[#164069218](https://www.pivotaltracker.com/story/show/164069218)

### CODE REVIEW

**Review by wasibani-roy — COMMENTED** [review]
Good work @CryceTruly for implementing the email verification feature. However when you look at the how to test section of your PR i believe you need to add abit more content such as after registering go to your email and click on the link sent to you via email upon clicking if successful proceed to login if succesful then the feature will be working just my thoughts on it

**Comment by CryceTruly** [comment]
> Good work @CryceTruly for implementing the email verification feature. However, when you look at the how to test section of your PR i believe you need to add a bit more content such as after registering to go to your email and click on the link sent to you via email upon clicking if successful proceed to login if successful then the feature will be working just my thoughts on it

Thanks, @wasibani-roy let me add that.

**Comment by Peace-Apple** [comment]
Good work chryce, however, In your PR description that part of relevant pivotal tracker stories could you make that story ID a link such that when someone clicks on it, they are taken directly to the story on pivotal tracker.

**Comment by CryceTruly** [comment]
> Good work chryce, however, In your PR description that part of relevant pivotal tracker stories could you make that story ID a link such that when someone clicks on it, they are taken directly to the story on pivotal tracker.

Thanks, @Peace-Apple, I have added that.

**Review by BagzieGracious — DISMISSED** [review]
@CryceTruly, thanks for the good work.

**Review by wasibani-roy — DISMISSED** [review]
Good work @CryceTruly just rebase your code because there conflicts it's having with develop

**Review by Peace-Apple — DISMISSED** [review]
Good work cryce, LGTM

**Review by wasibani-roy — DISMISSED** [review]
Good work @CryceTruly LGTM

**Review by BagzieGracious — DISMISSED** [review]
LGTM

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -76,9 +76,12 @@ class User(AbstractBaseUser, PermissionsMixin):
     # falsed.
     is_staff = models.BooleanField(default=False)
 
-    #The `is_verified` flag is expected to determine whose user account is
-    #verified by an email. For users who haved verified their emails the flag 
-    # will be false. 
+    # The `is_verified` flag is expected to determine whose user account is
```
Comment (marthamareal) [inline authors/apps/authentication/models.py:79]: @CryceTruly There are duplicates in the comments notes

File: authors/apps/authentication/serializers.py [hunk 2]
```diff
@@ -41,7 +41,9 @@ def validate_email(self, value):
 
         # Validate email has not been used to create account before
         if user_query_set.exists():
-            raise serializers.ValidationError("This email has already been used to create a user")
+            raise serializers.ValidationError(
+                "This email has already been used to create a user")
```
Comment (marthamareal) [inline authors/apps/authentication/serializers.py]: Also this formatting needs to be refactored

**Review by wasibani-roy — APPROVED** [review]
Good job on refactoring the validation part @CryceTruly LGTM

**Review by Peace-Apple — APPROVED** [review]
Good work Cryce, LGTM

**Review by BagzieGracious — APPROVED** [review]
Goodwork @CryceTruly, it truly sends a verification link

**Pull request merged**